core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -193,7 +196,8 @@ void *impl_allocate_common(const int device_id,
         KOKKOS_IMPL_CUDA_SAFE_CALL(cudaStreamSynchronize(stream));
       } else {
         Impl::cuda_device_synchronize(
-            "Kokkos::Cuda: backend fence after async malloc");
+            std::string("Kokkos::Cuda: backend fence after async malloc ") +
```

Comment:
Why did you change here and below every time we were passing a C string?

Reply:
What would be the better way to add to strings? The idea is to report the label for the respective allocation to Tools.